pydeseq2/utils.py

```diff
@@ -115,6 +115,28 @@ def load_data(
     return df
 
 
+def test_valid_counts(counts_df):
+    """Test that the count matrix contains valid inputs.
+
+    More precisely, test that inputs are non-negative integers.
+
+    Parameters
+    ----------
+    counts_df : pandas.DataFrame
+        Raw counts. One column per gene, rows are indexed by sample barcodes.
+    """
+    if counts_df.isna().any().any():
+        raise ValueError("NaNs are not allowed in the count matrix.")
```

Comment:
Good point. I've reproduced what DESeq2 does but it would make sense to throw a warning and output NaN results for genes that have at least one NaN in the count matrix.

I'll merge this PR for now and open an issue to discuss this.